### PR TITLE
Allow specifying `stdout` to Popen

### PR DESCRIPTION
**Pull request by s0undt3ch**

### CODE REVIEW

**Comment by s0undt3ch** [comment]
The reason behind this change is that on a particular session, we need access to the output, even if the command failed, for CI reporting reasons.

**Comment by theacodes** [comment]
Can you show me an example of when you'd use this and why the existing mechanisms aren't giving your the output you expect?

**Comment by mostealth** [comment]
When running in CI you might want to save pylint output to file (to be later parsed). pylint does not support outputting to file hence the only solution I found was to monkey patch subprocess.Popen so that I inject my own stdout/stderr file before calling session.run

**Comment by s0undt3ch** [comment]
What @mostealth said, that's our exact same scenario and why I created this PR.

This solution in use on our repo can be seen [here](https://github.com/saltstack/salt/blob/2018.3/noxfile.py#L819-L888)

File: nox/popen.py [hunk 1]
```diff
@@ -16,13 +16,11 @@
 import sys
 
 
-def popen(args, env=None, silent=False):
-    stdout = None
-
-    if silent:
+def popen(args, env=None, silent=False, stdout=None, stderr=subprocess.STDOUT):
```
Comment (theacodes) [inline nox/popen.py]: What happens when both `silent` and `stdout` are specified? What happens when the command fails but `stdout` isn't `PIPE`? Can we add test cases for that?

**Comment by theacodes** [comment]
That seems like a great reason. I had a comment about covering some edge cases, but I'm happy to merge this once we get that sorted. :)

**Comment by s0undt3ch** [comment]
Rebased and added two new tests based on the comments provided.

**Comment by theacodes** [comment]
Okay, I made `silent` and `stdout` mutually exclusive since `silent` doesn't have an effect if `stdout` is passed.

Does this still look good to you, @s0undt3ch? Also, it should be technically possible to override stderr now.  Do you mind adding a test for that?

**Comment by s0undt3ch** [comment]
Added tests for custom `stderr`

**Comment by theacodes** [comment]
Thank you, @s0undt3ch!

**Comment by s0undt3ch** [comment]
Anytime!
